scriptmodules/libretrocores/lr-scummvm.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="lr-scummvm"
+rp_module_desc="ScummVM port for libretro"
+rp_module_help="Copy your ScummVM games to $romdir/scummvm\n\nThe name of your game directories must be suffixed with '.svm' for direct launch in EmulationStation."
+rp_module_licence="GPL2 https://raw.githubusercontent.com/libretro/scummvm/master/COPYING"
+rp_module_section="exp"
+
+function sources_lr-scummvm() {
+    gitPullOrClone "$md_build" https://github.com/libretro/scummvm.git
+    # fix random freeze-on-quit (remove when patched upstream)
+    applyPatch "$md_data/01_fix_quit.diff"
+}
+
+function build_lr-scummvm() {
+    make clean -C backends/platform/libretro/build
+
+    local params=(HAVE_MT32EMU=1)
+    local platform
+    isPlatform "arm" && platform+="armv"
+    isPlatform "neon" && platform+="neon"
+    [[ -n "$platform" ]] && params+=(platform="$platform")
+
+    CXXFLAGS="$CXXFLAGS -DHAVE_POSIX_MEMALIGN=1" LDFLAGS="$LDFLAGS -shared -Wl,--no-undefined" \
+        make "${params[@]}" -C backends/platform/libretro/build
+    md_ret_require="$md_build/backends/platform/libretro/build/scummvm_libretro.so"
+}
+
+function install_lr-scummvm() {
+    md_ret_files=(
+        "backends/platform/libretro/build/scummvm_libretro.so"
+        "COPYING"
+    )
+}
+
+function configure_lr-scummvm() {
+    addEmulator 0 "romdir-$md_id" "scummvm" "$md_inst/retroarch-romdir.sh %ROM%"
+    addSystem "scummvm"
+    [[ "$md_mode" == "remove" ]] && return
+
+    # ensure rom dir and system retroconfig
+    mkRomDir "scummvm"
+    ensureSystemretroconfig "scummvm"
+
+    # download and extract auxiliary data (theme, extra)
+    downloadAndExtract "https://github.com/libretro/scummvm/raw/master/backends/platform/libretro/aux-data/scummvm.zip" "$biosdir"
+    chown -R $user:$user "$biosdir/scummvm"
+
+    # basic initial configuration (if config file not found)
+    if [[ ! -f "$biosdir/scummvm.ini" ]]; then
+        echo "[scummvm]" > "$biosdir/scummvm.ini"
+        iniConfig "=" "" "$biosdir/scummvm.ini"
+        iniSet "extrapath" "$biosdir/scummvm/extra"
+        iniSet "themepath" "$biosdir/scummvm/theme"
+        iniSet "soundfont" "$biosdir/scummvm/extra/Roland_SC-55.sf2"
+        iniSet "gui_theme" "scummmodern"
+        iniSet "subtitles" "true"
+        iniSet "multi_midi" "true"
+        iniSet "gm_device" "fluidsynth"
+        chown $user:$user "$biosdir/scummvm.ini"
+    fi
+
+    # enable speed hack core option if running in arm platform
+    isPlatform "arm" && setRetroArchCoreOption "scummvm_speed_hack" "enabled"
```

Comment:
is this needed for all arm platforms ? Maybe better for just armv6 or so ? I assume there are negative effects or it would be a default in the libretro core ?

Reply:
This is recommended by the lr-scummvm upstream docs (<https://docs.libretro.com/library/scummvm/>), hence I just followed it.

Their reasoning is this:

> This hack is considered 'safe' - games should run correctly when it is enabled, and most inaccuracies are imperceptible other than in a few edge cases. It remains a hack, though, and it is strongly recommended that users of desktop class machines keep it disabled.
> 
> However: For users of low power hardware (Android devices, single board computers), this hack is essentially mandatory for full speed operation of the core. For Android users in particular, the guides in the 'Usage' section of this document assume that the speed hack is enabled.

I myself and the testers in the forum has not noticed any problem with it.

Reply:
but does the core run at full speed on a rpi3 without it ? if so for example, we may want to leave it disabled.

Reply:
I will try with my set of games and come up with a conclusion. I didn't try further as they emphasize: *However: For users of low power hardware (Android devices, single board computers), this hack is essentially mandatory for full speed operation of the core.*